Assets/MixedRealityToolkit.Services/InputAnimation/InputAnimation.cs

```diff
@@ -0,0 +1,710 @@
+                return EvaluateHandJoint(time, joint, handJointCurvesRight);
+            }
+            else
+            {
+                return MixedRealityPose.ZeroIdentity;
+            }
+        }
+
+        /// Evaluate joint pose at the given time.
+        private MixedRealityPose EvaluateHandJoint(float time, TrackedHandJoint joint, Dictionary<TrackedHandJoint, PoseCurves> jointCurves)
+        {
+            if (jointCurves.TryGetValue(joint, out PoseCurves curves))
+            {
+                return EvaluatePose(curves, time);
+            }
+            else
+            {
+                return MixedRealityPose.ZeroIdentity;
+            }
+        }
+
+        /// <summary>
+        /// Evaluate the camera transform at the given time.
+        /// </summary>
+        public MixedRealityPose EvaluateCameraPose(float time)
+        {
+            return EvaluatePose(cameraCurves, time);
+        }
+
+        private static MixedRealityPose EvaluatePose(PoseCurves curves, float time)
+        {
+            float px = curves.PositionX.Evaluate(time);
+            float py = curves.PositionY.Evaluate(time);
+            float pz = curves.PositionZ.Evaluate(time);
+            float rx = curves.RotationX.Evaluate(time);
+            float ry = curves.RotationY.Evaluate(time);
+            float rz = curves.RotationZ.Evaluate(time);
+            float rw = curves.RotationW.Evaluate(time);
+
+            var pose = new MixedRealityPose();
+            pose.Position = new Vector3(px, py, pz);
+            pose.Rotation = new Quaternion(rx, ry, rz, rw);
+            pose.Rotation.Normalize();
+            return pose;
+        }
+
+        /// <summary>
+        /// Get the marker at the given index.
+        /// </summary>
+        public InputAnimationMarker GetMarker(int index)
+        {
+            return markers[index];
+        }
+
+        /// <summary>
+        /// Find an index i in the sorted events list, such that events[i].time &lt;= time &lt; events[i+1].time.
+        /// </summary>
+        /// <returns>
+        /// 0 &lt;= i &lt; eventCount if a full interval could be found.
+        /// -1 if time is less than the first event time.
+        /// eventCount-1 if time is greater than the last event time.
+        /// </returns>
+        /// <remarks>
+        /// Uses binary search.
+        /// </remarks>
+        public int FindMarkerInterval(float time)
+        {
+            int lowIdx = -1;
+            int highIdx = markers.Count;
+            while (lowIdx < highIdx - 1)
+            {
+                int midIdx = (lowIdx + highIdx) >> 1;
+                if (time >= markers[midIdx].time)
+                {
+                    lowIdx = midIdx;
+                }
+                else
+                {
+                    highIdx = midIdx;
+                }
+            }
+            return lowIdx;
+        }
+
+        /// Sort marker array by time values.
+        private void SortMarkers()
```

Comment:
Is this function still used? Doing a Ctrl+F on this page doesn't find anything

Reply:
Not used any more. Markers are only inserted one-by-one now and don't need global sorting. Removing.